### PR TITLE
app-misc/tmux: Fix pipe-pane unreliable.

### PR DESCRIPTION
**Pull request by strangemk2**
Bug: https://bugs.gentoo.org/640228
Package-Manager: Portage-2.3.13

### CODE REVIEW

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+inherit autotools flag-o-matic versionator
+
+DESCRIPTION="Terminal multiplexer"
+HOMEPAGE="http://tmux.github.io/"
+SRC_URI="https://github.com/${PN}/${PN}/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="alpha amd64 arm ~arm64 hppa ia64 ~mips ppc ppc64 ~s390 ~sh sparc x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
```
Comment (titanofold) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: ekeyword ~all tmux-2.3-r2

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+RDEPEND="${CDEPEND}
+	selinux? ( sec-policy/selinux-screen )
+	vim-syntax? (
+		|| (
+			app-editors/vim
+			app-editors/gvim
+		)
+	)"
+
+DOCS=( CHANGES FAQ README TODO )
```
Comment (SoapGentoo) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: none of these should be necessary with `einstalldocs`, nix

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	# usptream fixes (can be removed with next version bump)
+	"${FILESDIR}"/${P}-screen_write_copy-fix.patch
+	"${FILESDIR}"/${P}-pipe_pane-fix.patch
+)
+
+src_prepare() {
+	# bug 438558
+	# 1.7 segfaults when entering copy mode if compiled with -Os
+	replace-flags -Os -O2
```
Comment (SoapGentoo) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: do this in `src_configure` before calling `econf`

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+	# regenerate aclocal.m4 to support earlier automake versions
+	rm aclocal.m4 || die
```
Comment (SoapGentoo) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: do this after `default`

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	econf \
+		--sysconfdir="${EPREFIX}"/etc \
+		$(use_enable debug) \
+		$(use_enable utempter)
+}
+
+src_install() {
+	default
+
+	einstalldocs
```
Comment (SoapGentoo) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: not needed with `default`

File: app-misc/tmux/tmux-2.3-r2.ebuild [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	if ! version_is_at_least 1.9a ${REPLACING_VERSIONS:-1.9a}; then
+		echo
+		ewarn "Some configuration options changed in this release."
+		ewarn "Please read the CHANGES file in /usr/share/doc/${PF}/"
+		ewarn
+		ewarn "WARNING: After updating to ${P} you will _not_ be able to connect to any"
+		ewarn "older, running tmux server instances. You'll have to use an existing client to"
+		ewarn "end your old sessions or kill the old server instances. Otherwise you'll have"
+		ewarn "to temporarily downgrade to access them."
+		echo
```
Comment (SoapGentoo) [inline app-misc/tmux/tmux-2.3-r2.ebuild]: replace both this and the `echo` before with `ewarn`

Comment (strangemk2) [reply app-misc/tmux/tmux-2.3-r2.ebuild]: Do I suppose to delete these warnings?

Comment (SoapGentoo) [reply app-misc/tmux/tmux-2.3-r2.ebuild]: no? replace `echo` by `ewarn`

**Pull request closed without merging**